CardanoSharp.Wallet/Encoding/Bech32.cs

```diff
@@ -134,149 +124,10 @@ private byte[] ExpandHrp(string hrp)
-                }
-            }
-            else if (bits >= fromBits || (byte)((acc << (toBits - bits)) & maxv) != 0)
-            {
-                return null;
-            }
-            return result.ToArray();
-        }
-
-
-
-        private byte[] Bech32Decode(string bech32EncodedString, out string hrp)
-        {
-            bech32EncodedString = bech32EncodedString.ToLower();
-
-            int sepIndex = bech32EncodedString.LastIndexOf(Separator);
-            hrp = bech32EncodedString.Substring(0, sepIndex);
-            string data = bech32EncodedString.Substring(sepIndex + 1);
-
-            byte[] b32Arr = new byte[data.Length];
-            for (int i = 0; i < data.Length; i++)
-            {
-                b32Arr[i] = (byte)B32Chars.IndexOf(data[i]);
-            }
-
-            return b32Arr;
-        }
-
-
-        /// <summary>
-        /// Converts a bech-32 encoded string back to its byte array representation.
-        /// </summary>
-        /// <exception cref="FormatException"/>
-        /// <param name="bech32EncodedString">Bech-32 encoded string.</param>
-        /// <param name="witVer">Witness version</param>
-        /// <param name="hrp">Human readable part</param>
-        /// <returns>Byte array of the given string.</returns>
-        public byte[] Decode(string bech32EncodedString, out byte witVer, out string hrp)
-        {
-            byte[] b32Arr = Bech32Decode(bech32EncodedString, out hrp);
-            if (b32Arr.Length < CheckSumSize)
-            {
-                throw new FormatException("Invalid data length.");
-            }
-            if (!VerifyChecksum(hrp, b32Arr))
-            {
-                throw new FormatException("Invalid checksum.");
-            }
-
-            byte[] data = b32Arr.SubArray(0, b32Arr.Length - CheckSumSize);
-            byte[] b256Arr = ConvertBits(data, 5, 8, false);
-            if (b256Arr == null)
-            {
-                throw new FormatException("Invalid data format.");
-            }
-
-            witVer = b32Arr[0];
-            return b256Arr;
-        }
-
-
-        /// <summary>
-        /// Converts the given byte array to its equivalent string representation that is encoded with bech-32 digits,
-        /// with 6 byte appended checksum.
-        /// </summary>
-        /// <exception cref="ArgumentNullException"/>
-        /// <exception cref="ArgumentOutOfRangeException"/>
-        /// <param name="data">Byte array to encode</param>
-        /// <param name="hrp">Human readable part</param>
-        /// <returns>The string representation in bech-32 with a checksum.</returns>
-        public string Encode(byte[] data, string hrp)
-        {
-            if (data == null || data.Length == 0)
-                throw new ArgumentNullException(nameof(data), "Data can not be null or empty.");
-            if (!IsValidHrp(hrp))
-                throw new FormatException("Invalid HRP.");
```

Comment:
We might want to think about Custom Exceptions. There will be numerous time in our application that a witness, metadata, etc could be null so making some custom domain logic via Exceptions might work well for us